### PR TITLE
integrationTests: protocol rewards  and transaction fee rewards integ…

### PR DESCRIPTION
**Pull request by AdoAdoAdo**

### CODE REVIEW

File: integrationTests/testProcessorNodeWithMultisigner.go [hunk 1]
```diff
@@ -194,3 +197,99 @@ func DoConsensusSigningOnBlock(
...
+	map[uint32][]byte,
+) {
+
+	body := make(map[uint32]data.BodyHandler)
+	header := make(map[uint32]data.HeaderHandler)
+	consensusNodes := make(map[uint32][]*TestProcessorNode)
+	newRandomness := make(map[uint32][]byte)
+
+	// propose blocks
+	for i, _ := range nodesMap {
```
Comment (BeniaminDrasovean) [inline integrationTests/testProcessorNodeWithMultisigner.go]: Remove ,_

File: integrationTests/testProcessorNodeWithMultisigner.go [hunk 2]
```diff
@@ -194,3 +197,99 @@ func DoConsensusSigningOnBlock(
...
+		body[i], header[i], _, consensusNodes[i] = ProposeBlockWithConsensusSignature(i, nodesMap, round, nonce, prevRandomness[i])
+		newRandomness[i] = header[i].GetRandSeed()
+	}
+
+	// propagate blocks
+	for i, _ := range nodesMap {
```
Comment (BeniaminDrasovean) [inline integrationTests/testProcessorNodeWithMultisigner.go]: Remove ,_

File: integrationTests/testInitializer.go [hunk 3]
```diff
@@ -831,6 +845,51 @@ func GenerateSkAndPkInShard(
...
+	nodes map[uint32][]*TestProcessorNode,
+	nbTxsPerShard int,
+	gasPricePerTx uint64,
+	gasLimitPerTx uint64,
+	valueToTransfer *big.Int,
+) map[uint32][]crypto.PrivateKey {
+	sendersPrivKeysMap := make(map[uint32][]crypto.PrivateKey)
+	receiversPubKeysMap := make(map[uint32][]crypto.PublicKey)
+
+	for shardId, _ := range nodes {
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply integrationTests/testInitializer.go]: done

File: integrationTests/testProcessorNodeWithMultisigner.go (hunk 1 above)
Comment (AdoAdoAdo) [reply integrationTests/testProcessorNodeWithMultisigner.go]: done

File: integrationTests/testProcessorNodeWithMultisigner.go (hunk 2 above)
Comment (AdoAdoAdo) [reply integrationTests/testProcessorNodeWithMultisigner.go]: done

File: integrationTests/testInitializer.go [hunk 4]
```diff
@@ -707,7 +707,7 @@ func DisplayAndStartNodes(nodes []*TestProcessorNode) {
 func GenerateAndDisseminateTxs(
 	n *TestProcessorNode,
 	senders []crypto.PrivateKey,
-	receiversPrivateKeys map[uint32][]crypto.PrivateKey,
+	receiversPublicKeysMap map[uint32][]crypto.PublicKey,
```
Comment (iulianpascalau) [inline integrationTests/testInitializer.go]: :+1:

File: integrationTests/testInitializer.go [hunk 5]
```diff
@@ -819,6 +828,11 @@ func GenerateSkAndPkInShard(
 	keyGen := signing.NewKeyGenerator(suite)
 	sk, pk := keyGen.GeneratePair()
 
+	if shardId == sharding.MetachainShardId {
+		// for metachain generate in shard 0
+		shardId = 0
```
Comment (iulianpascalau) [inline integrationTests/testInitializer.go]: :+1:

File: integrationTests/testInitializer.go [hunk 6]
```diff
@@ -1174,13 +1233,18 @@ func PubKeysMapFromKeysMap(keyPairMap map[uint32][]*TestKeyPair) map[uint32][]st
...
 // GenValidatorsFromPubKeys generates a map of validators per shard out of public keys map
-func GenValidatorsFromPubKeys(pubKeysMap map[uint32][]string) map[uint32][]sharding.Validator {
+func GenValidatorsFromPubKeys(pubKeysMap map[uint32][]string, nbShards uint32) map[uint32][]sharding.Validator {
 	validatorsMap := make(map[uint32][]sharding.Validator)
 
 	for shardId, shardNodesPks := range pubKeysMap {
 		shardValidators := make([]sharding.Validator, 0)
+		shardCoordinator, _ := sharding.NewMultiShardCoordinator(nbShards, shardId)
 		for i := 0; i < len(shardNodesPks); i++ {
-			address := []byte(shardNodesPks[i][:32])
```
Comment (iulianpascalau) [inline integrationTests/testInitializer.go]: :+1:

File: integrationTests/testProcessorNodeWithMultisigner.go [hunk 7]
```diff
@@ -194,3 +197,99 @@ func DoConsensusSigningOnBlock(
...
+	t *testing.T,
+	headers map[uint32]data.HeaderHandler,
+	nodesMap map[uint32][]*TestProcessorNode,
+) {
+	var v interface{}
+	var ok bool
+
+	// all nodes in metachain have the block headers in pool as interceptor validates them
+	for shHeader, header := range headers {
+		headerBytes, _ := TestMarshalizer.Marshal(header)
```
Comment (iulianpascalau) [inline integrationTests/testProcessorNodeWithMultisigner.go]: might use core.CalculateHash func

File: integrationTests/multiShard/block/executingRewardMiniblocks_test.go [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/integrationTests"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestExecuteBlocksWithOnlyRewards(t *testing.T) {
```
Comment (iulianpascalau) [inline integrationTests/multiShard/block/executingRewardMiniblocks_test.go]: WithOnlyRewards doesn't sound too good. Might rename this to "TestExecuteBlocksWithTransactionsAndCheckRewards"
Might have another test for blocks without transactions.

File: integrationTests/multiShard/block/executingRewardMiniblocks_test.go [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+	var consensusNodes map[uint32][]*integrationTests.TestProcessorNode
+	mapRewardsForAddress := make(map[string]uint32)
+	nbTxsForLeaderAddress := make(map[string]uint32)
+
+	gasPrice := uint64(10)
+	gasLimit := uint64(100)
+	feePerTxForLeader := gasPrice * gasLimit / 2
+	valToTransfer := big.NewInt(100)
+
+	sendersPrivateKeys := integrationTests.CreateAndSendIntraShardTransactions(
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply integrationTests/multiShard/block/executingRewardMiniblocks_test.go]: right, now balance is checked on interceptors.

File: integrationTests/multiShard/block/executingRewardMiniblocks_test.go [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+func getBlockProposersIndexes(
+	consensusMap map[uint32][]*integrationTests.TestProcessorNode,
+	nodesMap map[uint32][]*integrationTests.TestProcessorNode,
+) map[uint32]int {
+
+	indexProposer := make(map[uint32]int)
+
+	for sh, testNodeList := range nodesMap {
+		for k, testNode := range testNodeList {
+			if reflect.DeepEqual(consensusMap[sh][0], testNode) {
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply integrationTests/multiShard/block/executingRewardMiniblocks_test.go]: done

[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply integrationTests/multiShard/block/executingRewardMiniblocks_test.go]: done

File: integrationTests/multiShard/block/executingRewardMiniblocks_test.go [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+
+	return nbTxs
+}
+
+func updateExpectedRewards(rewardsForAddress map[string]uint32, addresses []string) {
+	for i := 0; i < len(addresses); i++ {
+		if addresses[i] == "" {
+			continue
+		}
+		currentRewards, ok := rewardsForAddress[addresses[i]]
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply integrationTests/multiShard/block/executingRewardMiniblocks_test.go]: done

File: integrationTests/multiShard/block/executingRewardMiniblocks_test.go [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+	t *testing.T,
+	transactionsForLeader map[string]uint32,
+	addressProposer string,
+	nbTransactions uint32,
+) {
+	if addressProposer == "" {
+		assert.Error(t, errors.New("invalid address"))
+	}
+
+	proposedTransactions, ok := transactionsForLeader[addressProposer]
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply integrationTests/multiShard/block/executingRewardMiniblocks_test.go]: done

File: integrationTests/testProcessorNodeWithMultisigner.go (hunk 7 above)
Comment (AdoAdoAdo) [reply integrationTests/testProcessorNodeWithMultisigner.go]: done

File: integrationTests/multiShard/block/executingRewardMiniblocks_test.go (hunk 8 above)
Comment (AdoAdoAdo) [reply integrationTests/multiShard/block/executingRewardMiniblocks_test.go]: done

**Pull request merged**